### DISCUSSION TITLE
Sotu kiinni useammalla asiakkaalla

### DISCUSSION THREAD

**Discussion by emta001** [KohaSuomi/Koha#1136] [Tarkistuspyynnöt]
Ainakin Lumpeissa ja OUTIssa löytyy asiakkaita, joilla on sama sotu-arvo borrower_attributes-taulussa. Ainakin OUTIssa näille asiakkaille on laitettu huomautus tupla-arvosta (joissakin tapauksissa huomautus on vain toisella asiakkaalla). Seuraavalla Annelin taikarimpsulla saa kiinni kaikki sotut, jotka ovat kiinni kahdessa tai useammassa asiakkaassa.

```
SELECT
ba.attribute,ba.borrowernumber,dateenrolled,lastseen
FROM borrower_attributes ba
LEFT JOIN borrowers b ON ba.borrowernumber=b.borrowernumber
INNER JOIN
(SELECT attribute
   FROM borrower_attributes
   WHERE code = "SSN"
GROUP BY attribute
HAVING COUNT(*)>1) as tuplat ON ba.attribute = tuplat.attribute
ORDER BY 1
```

HUOM! Raportti on hidas!

**Comment by pknuuti** [comment]
Näyttää noita olevan Vaarassakin yli 200. Tarkistin kolme ensimmäistä paria ja kaikki ovat keskenään samoja, henkilötiedoissa vain heittoja nimen täydellisyydessä (toisessa vain yksi etunimi, toisessa kaksi). Mitä näille pitäisi tehdä?

**Reply by emta001** [reply]
Ainakin OUTIssa asiakkaille on lisätty viesti "Sotu kiinni useammassa asiakkaassa" ja ohjeistettu sitten tarkistamaan, kumpi on oikea asiakas, kun asiakas sattuu paikalle. Ehkäpä se on hyvä toimintatapa, jos ei voida olla varmoja, kumpi asiakkaista pitää poistaa.

**Comment by ReettaSiili** [comment]
Siili: none

**Comment by Tuomas-Kyyti** [comment]
Kyytissä 40 kpl. Kaikki asiakkaat siirtyneet edellisestä järjestelmästä. Vain kolmella oli lastseen-arvo (lisäsin sen sarakkeen raporttiin). Käytännössä asiakkaitten siivousajo korjaa tilanteen sitten kun se tehdään.

**Comment by ElinaUotila** [comment]
Kirkesistä ei löytynyt tällaisia asiakkaita.

**Comment by PiaKusmin** [comment]
Lapissa tuplia on runsaasti. Kaikki todennäköisesti konversioiden aikaisia eli vanhoja. Siivotaan pois, kun keretään.

**Comment by Hanna-Lumme** [comment]
Lumpeissa näitä on tosiaan ihan kiitettävästi. Siivotaan jäljelle jääneet pois, kunhan ollaan tehty tuo asiakasrekisterin puhdistusajo muutaman viikon päästä.

**Comment by pirliis** [comment]
OUTIssa 2049 asiakasta. Ei aleta siivoamaan käsin. Odotetaan, jos saisimme vanhentuneiden asiakastietueiden poistoajon jossain vaiheessa, jossa varmaan suurin osa poistuu.

**Comment by sksandel** [comment]
Vaskissa ei ollut yhtään.
